mesonbuild/compilers/cuda.py

```diff
@@ -60,23 +61,423 @@ def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoic
+            '--machine':                            '-m',                           # 4.2.3.19
+            '--compiler-options':                   '-Xcompiler',                   # 4.2.4.1
+            '--linker-options':                     '-Xlinker',                     # 4.2.4.2
+            '--archive-options':                    '-Xarchive',                    # 4.2.4.3
+            '--ptxas-options':                      '-Xptxas',                      # 4.2.4.4
+            '--nvlink-options':                     '-Xnvlink',                     # 4.2.4.5
+            '--threads':                            '-t',                           # 4.2.5.4
+            '--keep-dir':                           '-keep-dir',                    # 4.2.5.8
+            '--run-args':                           '-run-args',                    # 4.2.5.11
+            '--input-drive-prefix':                 '-idp',                         # 4.2.5.12
+            '--dependency-drive-prefix':            '-ddp',                         # 4.2.5.13
+            '--drive-prefix':                       '-dp',                          # 4.2.5.14
+            '--dependency-target-name':             '-MT',                          # 4.2.5.15
+            '--default-stream':                     '-default-stream',              # 4.2.6.1
+            '--gpu-architecture':                   '-arch',                        # 4.2.7.1
+            '--gpu-code':                           '-code',                        # 4.2.7.2
+            '--generate-code':                      '-gencode',                     # 4.2.7.3
+            '--relocatable-device-code':            '-rdc',                         # 4.2.7.4
+            '--entries':                            '-e',                           # 4.2.7.5
+            '--maxrregcount':                       '-maxrregcount',                # 4.2.7.6
+            '--ftz':                                '-ftz',                         # 4.2.7.8
+            '--prec-div':                           '-prec-div',                    # 4.2.7.9
+            '--prec-sqrt':                          '-prec-sqrt',                   # 4.2.7.10
+            '--fmad':                               '-fmad',                        # 4.2.7.11
+            '--Werror':                             '-Werror',                      # 4.2.8.9
+            '--diag-error':                         '-diag-error',                  # 4.2.8.11
+            '--diag-suppress':                      '-diag-suppress',               # 4.2.8.12
+            '--diag-warn':                          '-diag-warn',                   # 4.2.8.13
+            '--options-file':                       '-optf',                        # 4.2.8.17
+            '--time':                               '-time',                        # 4.2.8.18
+            '--qpp-config':                         '-qpp-config',                  # 4.2.8.19
+        }
```

Comment:
I would avoid creating a huge dict like this inside a function body, since these will be recreated each time the function is called. These should be module or class level private.